### ISSUE TITLE
Testing scaling

### ISSUE THREAD

**Issue by NiruBhai** [code-mphi/ECOGEN#23]
Dear all,

I have been trying to test strong scaling of ECOGEN in an HPC. The tutorials used were the spherical collapse and non spherical collapse cases. But the [results](https://docs.google.com/spreadsheets/d/1whQbIwpz1Nk0EyBeW8MvzR1dGZ4oQwUsdMB6Xwe6nmQ/edit?usp=sharing) are unexpectedly bad and I am confused. I went through the main journal article of ECOGEN and found some examples used for scaling but I am not sure whether they are available as such. Could anyone please point out if they are there or provide the input files if not there?

Thanking you
Yours sincerely

Niranjan K. Sabu

**Comment by kevinschmidmayer** [comment]
Hi Niranjan,

Are you using AMR for your tests? Because AMR is not scaling well for sure (as shown and explained in the papers). If you don't use AMR, so a Cartesian grid, you should expect good scaling whatever the test case. Also, what is the number of cells per core you are using? And what is the total number of cores/nodes? Don't expect a good scaling for a small number of cells per core and for a large number of cores (superior to a few thousands).

Best regards,
Kevin

**Comment by NiruBhai** [comment]
The mesh size was 100 * 100 * 100 cells and I tried with the number of cores ranging from 16 to 192, with 16 core/node, 24 core/node as well as 32 core/node (Max limit is 40 core/node). I did not use AMR as well.

**Comment by NiruBhai** [comment]
@kevinschmidmayer could you please point out the test cases for scaling given in https://doi.org/10.1016/j.cpc.2019.107093 or give corresponding input files so that I can run and compare my own results with the ones given in the article? That would confirm whether its a problem from the HPC's side or not.

**Comment by kevinschmidmayer** [comment]
Since I don't have the specific files I used for the paper, I just quickly tried with another 3D test case and it worked as expected for me. Though I don't have access to a supercomputer right now so I did not try with a lot of cores but I am not expecting anything different from previous studies. Specifically that we didn't touch the parallel part of the program since then, at least for Cartesian non-AMR meshes. I am also not expecting anything different for whatever test case you use.

The test case I ran is using the PUEq model: libTests/referenceTestCases/PUEq/3D/shockBubble/heliumAir/
I modified the mainV5.xml file to activate the time control with the number of iterations and I setup 20 iterations. I also modified the meshV5.xml file to deactivate the AMR and refine the mesh to 400x80x80 in order to have a decent number of cells per core.

Hope it helps.
